server/graphql/v1/mutations/collectives.js

```diff
@@ -462,8 +463,19 @@ export async function archiveCollective(_, args, req) {
     return updatedCollective;
   }
 
-  // TODO: cascade deactivation to EVENTs and PROJECTs?
+  const projects = await collective.getProjects();
+  const events = await collective.getEvents();
 
+  if (projects && projects.length) {
+    projects.forEach(async project => {
+      await project.update({ isActive: false, deactivatedAt: Date.now(), approvedAt: null, HostCollectiveId: null });
+    });
```

Comment:
We're still not awaiting here. Try this:

```es6
const testPromise = new Promise(resolve => setTimeout(() => resolve(console.log('Done!')), 5000));
[42].forEach(async () => await testPromise());
console.log('Called promises');
```

You'll notice that "Called promises" will be displayed before "Done!".

We need to use `Promise.all` or a similar mechanism.

Reply:
You are right. Forgot that I should be using PromiseAll or new ES6 for...of instead. Commiting...